src/node.rs

```diff
@@ -773,6 +866,57 @@ impl<'a, V: 'a> Iterator for Iter<'a, V> {
     }
 }
 
+/// A mutable iterator which traverses the nodes in a tree, in depth first order.
+///
+/// The first element of an item is the level of the traversing node.
+#[derive(Debug)]
+pub struct IterMut<'a, V: 'a> {
+    stack: Vec<(usize, &'a mut Node<V>)>,
+}
+
+/// A reference to an immediate node (without child or sibling) with its
+/// label and a mutable reference to its value, if present.
+pub struct NodeMutRef<'a, V: 'a> {
```

Comment:
IMO, we can remove the `Ref` suffix from the class name.
```suggestion
pub struct NodeMut<'a, V: 'a> {
```

Reply:
I've done so 👍🏾